src/Bus/MongoBatchRepository.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace MongoDB\Laravel\Bus;
+
+use Carbon\CarbonImmutable;
+use Closure;
+use DateTimeInterface;
+use Illuminate\Bus\Batch;
+use Illuminate\Bus\BatchFactory;
+use Illuminate\Bus\DatabaseBatchRepository;
+use Illuminate\Bus\PendingBatch;
+use Illuminate\Bus\PrunableBatchRepository;
+use Illuminate\Bus\UpdatedBatchJobCounts;
+use Illuminate\Support\Carbon;
+use MongoDB\BSON\ObjectId;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\Driver\ReadPreference;
+use MongoDB\Laravel\Collection;
+use MongoDB\Laravel\Connection;
+use MongoDB\Operation\FindOneAndUpdate;
+use Override;
+
+use function date_default_timezone_get;
+use function is_string;
+use function serialize;
+use function unserialize;
+
+// Extending DatabaseBatchRepository is necessary so methods pruneUnfinished and pruneCancelled
+// are called by PruneBatchesCommand
+class MongoBatchRepository extends DatabaseBatchRepository implements PrunableBatchRepository
+{
+    private Collection $collection;
+
+    public function __construct(
+        BatchFactory $factory,
+        Connection $connection,
+        string $collection,
+    ) {
+        $this->collection = $connection->getCollection($collection);
+
+        parent::__construct($factory, $connection, $collection);
+    }
+
+    #[Override]
+    public function get($limit = 50, $before = null): array
+    {
+        if (is_string($before)) {
+            $before = new ObjectId($before);
+        }
+
+        return $this->collection->find(
+            $before ? ['_id' => ['$lt' => $before]] : [],
+            [
+                'limit' => $limit,
+                'sort' => ['_id' => -1],
+                'typeMap' => ['root' => 'array', 'document' => 'array', 'array' => 'array'],
+            ],
+        )->toArray();
+    }
+
+    #[Override]
+    public function find(string $batchId): ?Batch
+    {
+        $batchId = new ObjectId($batchId);
+
+        $batch = $this->collection->findOne(
+            ['_id' => $batchId],
+            [
+                'readPreference' => new ReadPreference(ReadPreference::PRIMARY),
```

Comment:
Why does `find()` use an explicit read preference, while `get()` does not? This may warrant a comment.

Reply:
That definitely seems worth of a comment that mentions the SQL version using a writable connection.